src/Loop/Driver.php

```diff
@@ -285,4 +311,35 @@
      * @return null|object|resource The loop handle the event loop operates on. `null` if there is none.
      */
     abstract public function getHandle();
+
+    /**
+     * Create a watcher identifier.
+     *
+     * Any watcher identifiers returned to clients by the driver MUST be created using this method. The returned watcher
+     * identifier SHALL be unique for the duration of the PHP process.
+     *
+     * @return string The watcher identifier.
+     */
+    final protected function createWatcherId()
+    {
+        return $this->watcherIdPrefix . $this->nextWatcherId++;
+    }
+
+    /**
+     * Validate a watcher identifier which was passed to the driver by a client.
+     *
+     * The driver MUST call this method when it is provided with a watcher identifier which it does not recognise. The
+     * driver SHOULD NOT call this method every time it receives a watcher identifier, as doing so would be relatively
+     * expensive and is not typically necessary.
+     *
+     * @param string $watcherId The watcher identifier.
+     *
+     * @throws InvalidWatcherException If the watcher identifier does not belong to this driver.
+     */
+    final protected function validateWatcherId($watcherId)
+    {
+        if (0 !== \strpos($watcherId, $this->watcherIdPrefix)) {
+            throw new InvalidWatcherException($watcherId, 'A watcher was passed to the wrong driver.');
```

Comment:
The exception message should contain the `watcherId` here, so e.g. `"Watcher '{$watcherId}' has been passed to the wrong driver. Watchers passed to this driver must start with '{$this->watcherIdPrefix}'."`.